homeassistant/components/prosegur/config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Config flow for Prosegur Alarm integration."""
+import logging
+
+from pyprosegur.auth import COUNTRY, Auth
+from pyprosegur.installation import Installation
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_COUNTRY, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required("country"): vol.In(COUNTRY.keys()),
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+    try:
+        session = aiohttp_client.async_get_clientsession(hass)
+        auth = Auth(
+            session, data[CONF_USERNAME], data[CONF_PASSWORD], data[CONF_COUNTRY]
+        )
+        install = await Installation.retrieve(auth)
+    except ConnectionRefusedError:
+        raise InvalidAuth from ConnectionRefusedError
```

Comment:
Where did you define `InvalidAuth`? You could also just catch `ConnectionRefusedError` and `ConnectionError` in your try/catch in `async_step_user`. I don't see why you would need to redefine exceptions, when your library is already throwing proper exceptions.

Reply:
I'm just using the code generated from the scaffold script 😅 

I figure the exceptions should inherit from exceptions.HomeAssistantError so I'm translating from the library to the component

Reply:
I am not 100% sure on this point, so would be good to leave this open for another reviewer.

However, I would say that you only need to define a specific exception when your library doesn't throw proper exceptions. But let's see if a core member agrees to this.

Reply:
Your current way looks broken though, since you are raising Exceptions that you didn't define or import.

Reply:
But these exception are defined in `config_flow.py` (last lines)

Reply:
I guess I was looking at a specific commit view, I thought you removed them during your refactoring.. :-)